sdks/python/apache_beam/examples/snippets/transforms/elementwise/runinference_test.py

```diff
@@ -0,0 +1,125 @@
+  assert_matches_stdout(actual, expected)
+
+
+def check_torch_unkeyed_model_handler():
+  expected = '''[START torch_unkeyed_model_handler]
+PredictionResult(example=tensor([10.]), inference=tensor([52.2325], grad_fn=<UnbindBackward>))
+PredictionResult(example=tensor([40.]), inference=tensor([201.1165], grad_fn=<UnbindBackward>))
+PredictionResult(example=tensor([60.]), inference=tensor([300.3724], grad_fn=<UnbindBackward>))
+PredictionResult(example=tensor([90.]), inference=tensor([449.2563], grad_fn=<UnbindBackward>))
+[END torch_unkeyed_model_handler] '''.splitlines()[1:-1]
+  return expected
+
+
+def check_sklearn_unkeyed_model_handler(actual):
+  expected = '''[START sklearn_unkeyed_model_handler]
+PredictionResult(example=array([20.], dtype=float32), inference=array([100.], dtype=float32))
+PredictionResult(example=array([40.], dtype=float32), inference=array([200.], dtype=float32))
+PredictionResult(example=array([60.], dtype=float32), inference=array([300.], dtype=float32))
+PredictionResult(example=array([90.], dtype=float32), inference=array([450.], dtype=float32))
+[END sklearn_unkeyed_model_handler]  '''.splitlines()[1:-1]
+  assert_matches_stdout(actual, expected)
+
+
+@mock.patch('apache_beam.Pipeline', TestPipeline)
+@mock.patch(
+    'apache_beam.examples.snippets.transforms.elementwise.runinference.print',
+    str)
+class RunInferenceTest(unittest.TestCase):
+  def test_sklearn_unkeyed_model_handler(self):
+    runinference.sklearn_unkeyed_model_handler(
+        check_sklearn_unkeyed_model_handler)
+
+  def test_sklearn_keyed_model_handler(self):
+    runinference.sklearn_keyed_model_handler(check_sklearn_keyed_model_handler)
+
+
+@mock.patch('apache_beam.Pipeline', TestPipeline)
+@mock.patch('sys.stdout', new_callable=StringIO)
+class RunInferenceStdoutTest(unittest.TestCase):
+  @pytest.mark.uses_pytorch
+  def test_check_torch_keyed_model_handler(self, mock_stdout):
+    runinference.torch_keyed_model_handler()
+    predicted = mock_stdout.getvalue().splitlines()
+    expected = check_torch_keyed_model_handler()
+    actual_stdout = [line.split(':')[0] for line in predicted]
+    replace_fn = lambda x: re.sub(r"<UnbindBackward\d*>", "<UnbindBackward>", x)
+    actual_stdout = [replace_fn(x) for x in actual_stdout]
+    expected_stdout = [line.split(':')[0] for line in expected]
+    self.assertEqual(actual_stdout, expected_stdout)
+
+  @pytest.mark.uses_pytorch
+  def test_check_torch_unkeyed_model_handler(self, mock_stdout):
+    runinference.torch_unkeyed_model_handler()
+    predicted = mock_stdout.getvalue().splitlines()
+    expected = check_torch_unkeyed_model_handler()
+    actual_stdout = [line.split(':')[0] for line in predicted]
+    replace_fn = lambda x: re.sub(r"<UnbindBackward\d*>", "<UnbindBackward>", x)
+    actual_stdout = [replace_fn(x) for x in actual_stdout]
+    expected_stdout = [line.split(':')[0] for line in expected]
+    self.assertEqual(actual_stdout, expected_stdout)
```

Comment:
Yes, but this is only for this particular torch test because
1. on my system, the output of the pipeline is almost same except it prints `UnbindBackward0` but on Jenkins it prints `UnbindBackward` so I had to remove any trailing numbers using regex. This is only for torch test. Thats why I had to parse the output. 
2. Sklearn tests don't follow that case. So I created simple test cases for them. Thats why torch and sklearn are in different classes.